src/lib/mina_base/zkapp_command.ml

```diff
@@ -1185,34 +1191,43 @@ end = struct
                     return err
               in
               if Control.(Tag.equal Tag.Proof (Control.tag p.authorization))
-              then (
+              then
```

Comment:
Would it be better `Applied`|`Failed` check is performed before entering this block? That way none of VKs are included. Currently there could be some VKs included and some not and that might cause confusion.